hazelcast/src/main/java/com/hazelcast/internal/ascii/rest/HttpPostCommandProcessor.java

```diff
@@ -260,11 +264,26 @@ private void handleMap(HttpPostCommand command, String uri) {
         }
         String mapName = uri.substring(URI_MAPS.length(), indexEnd);
         String key = uri.substring(indexEnd + 1);
-        byte[] data = command.getData();
-        textCommandService.put(mapName, key, new RestValue(data, command.getContentType()), -1);
+        Object value = extractValueObjectFromData(command);
+        textCommandService.put(mapName, key, value, -1);
         command.send200();
     }
 
+    @NotNull
+    private static Object extractValueObjectFromData(HttpPostCommand command) {
+        byte[] data = command.getData();
+        if (JSON_CONTENT_TYPE.equals(command.getMediaType())) {
+            Charset charset = command.getCharset();
+            if (charset == null) {
+                charset = StandardCharsets.UTF_8;
```

Comment:
this is the default encoding used for JSON: https://datatracker.ietf.org/doc/html/rfc4627#section-3